### PR TITLE
Remove session from SmartID authentication

### PR DESCRIPTION
**Pull request by v3rm0n**

### CODE REVIEW

File: src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
+import javax.servlet.http.HttpSession;
 import org.springframework.stereotype.Component;
-import org.springframework.web.context.annotation.SessionScope;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
 
 @Component
-@SessionScope
-public class GenericSessionStore implements Serializable {
+public class GenericSessionStore {
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java:13]: What changed here? Not a session scoped bean anymore?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -40,6 +40,7 @@ spring:
     store-type: none
     jdbc:
       initialize-schema: never
+  cloud.mvc.token-relay.enabled: false
```
Comment (ErkoRisthein) [inline src/main/resources/application.yml:43]: What does it affect?

File: src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java (hunk 1 above)
Comment (v3rm0n) [reply src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java:13]: Yes, injecting a session scoped bean will create a session automatically which we do not want

File: src/main/resources/application.yml (hunk 2 above)
Comment (v3rm0n) [reply src/main/resources/application.yml:43]: It uses a session so it forces one to be created. It didn't seem to be used anywhere, at least downstream OAuth2 calls still work fine even when this is disabled.

**Review by ErkoRisthein — APPROVED** [review]

**Pull request merged**